microsoft/CBL-Mariner
# Adding python module pyrpm spec

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*

---

###### Summary 

The script ./toolkit/scripts/license_map.py makes use of the python module pyrpm.spec which does not appear to be available on Mariner

###### Change Log  



- Change
- Change
- Change

###### Does this affect the toolchain?  



**YES/NO**

###### Test Methodology

- Built and installed rpm:
```
azaugg@azaugg-ld77 [ ~/rpmbuild ]$ rpmbuild -bb SPECS/python-rpm-spec.spec
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.KMlnMr
+ umask 022
+ cd /home/azaugg/rpmbuild/BUILD
+ /usr/lib/rpm/mariner/gen-ld-script.sh python-rpm-spec 0.14.1 /home/azaugg/rpmbuild '%{mariner_release_version}'
gen-ld-script.sh name(python-rpm-spec) version(0.14.1) _topdir(/home/azaugg/rpmbuild) osversion(%{mariner_release_version})
==== ELF note generator v2.1.2 ====
outdir is: /home/azaugg/rpmbuild/BUILD/
Stamp method: LinkerOnly

#ifndef _AUTO_MODULE_INFO_H_
#define _AUTO_MODULE_INFO_H_

#define MODULE_VERSION      "0.14.1.0"
#define PACKAGE_VERSION     "0.14.1"
#define PACKAGE_NAME        "python-rpm-spec"
#define TARGET_OS           "mariner"
#define TARGET_OS_VERSION   "%{mariner_release_version}"

#endif //_AUTO_MODULE_INFO_H_
Generated /home/azaugg/rpmbuild/BUILD/auto_module_info.h file...

Successfully generated module_info.ld and .note.package files...
+ cd /home/azaugg/rpmbuild/BUILD
+ rm -rf python-rpm-spec-0.14.1
+ /usr/lib/rpm/rpmuncompress -x /home/azaugg/rpmbuild/SOURCES/python-rpm-spec-v0.14.1.tar.gz
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd python-rpm-spec-0.14.1
+ /bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ RPM_EC=0
++ jobs -p
+ exit 0
Executing(%install): /bin/sh -e /var/tmp/rpm-tmp.RwuPKS
+ umask 022
+ cd /home/azaugg/rpmbuild/BUILD
+ /bin/rm -rf /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64
+ /bin/mkdir -p /home/azaugg/rpmbuild/BUILDROOT
+ /bin/mkdir /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64
+ cd python-rpm-spec-0.14.1
+ rm -rf /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64
+ mkdir -p /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64//usr/lib/python3.9/site-packages/
+ cp -r pyrpm /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64//usr/lib/python3.9/site-packages/
+ '[' '%{buildarch}' = noarch ']'
+ QA_CHECK_RPATHS=1
+ case "${QA_CHECK_RPATHS:-}" in
+ /usr/lib/rpm/check-rpaths
+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/brp-compress
+ /usr/lib/rpm/brp-strip-comment-note /bin/strip /bin/objdump
+ /usr/lib/rpm/brp-strip-static-archive /bin/strip
+ /usr/lib/rpm/mariner/brp-python-bytecompile /usr/bin/python3 1 0
Bytecompiling .py files below /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/lib/python3.9 using /usr/bin/python3.9
+ find /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64 -name '*.pc'
+ xargs '-I{}' sed -i -e 's@-Wl,-dT,/home/azaugg/rpmbuild/BUILD/module_info.ld@ @' '{}'
+ /usr/lib/rpm/brp-strip /bin/strip
+ /usr/lib/rpm/brp-strip-debug-symbols /bin/strip
+ /usr/lib/rpm/brp-strip-unneeded /bin/strip
Processing files: python3-rpm-spec-0.14.1-1.cm2.noarch
Executing(%doc): /bin/sh -e /var/tmp/rpm-tmp.XFo6lG
+ umask 022
+ cd /home/azaugg/rpmbuild/BUILD
+ cd python-rpm-spec-0.14.1
+ DOCDIR=/home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/share/doc/python3-rpm-spec
+ export LC_ALL=C
+ LC_ALL=C
+ export DOCDIR
+ /bin/mkdir -p /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/share/doc/python3-rpm-spec
+ cp -pr README.md /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/share/doc/python3-rpm-spec
+ RPM_EC=0
++ jobs -p
+ exit 0
Executing(%license): /bin/sh -e /var/tmp/rpm-tmp.eC71ZL
+ umask 022
+ cd /home/azaugg/rpmbuild/BUILD
+ cd python-rpm-spec-0.14.1
+ LICENSEDIR=/home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/share/licenses/python3-rpm-spec
+ export LC_ALL=C
+ LC_ALL=C
+ export LICENSEDIR
+ /bin/mkdir -p /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/share/licenses/python3-rpm-spec
+ cp -pr LICENSE /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64/usr/share/licenses/python3-rpm-spec
+ RPM_EC=0
++ jobs -p
+ exit 0
Provides: python3-rpm-spec = 0.14.1-1.cm2
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Requires: python(abi) = 3.9
Checking for unpackaged file(s): /usr/lib/rpm/check-files /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64
Wrote: /home/azaugg/rpmbuild/RPMS/noarch/python3-rpm-spec-0.14.1-1.cm2.noarch.rpm
Executing(%clean): /bin/sh -e /var/tmp/rpm-tmp.og3xco
+ umask 022
+ cd /home/azaugg/rpmbuild/BUILD
+ cd python-rpm-spec-0.14.1
+ /bin/rm -rf /home/azaugg/rpmbuild/BUILDROOT/python-rpm-spec-0.14.1-1.cm2.x86_64
+ RPM_EC=0
++ jobs -p
+ exit 0
Executing(rmbuild): /bin/sh -e /var/tmp/rpm-tmp.uKRHa3
+ umask 022
+ cd /home/azaugg/rpmbuild/BUILD
+ rm -rf python-rpm-spec-0.14.1 python-rpm-spec-0.14.1.gemspec
+ RPM_EC=0
++ jobs -p
+ exit 0
azaugg@azaugg-ld77 [ ~/rpmbuild ]$
```
- Then installed the rpm
```
azaugg@azaugg-ld77 [ ~/rpmbuild ]$ sudo rpm -i RPMS/noarch/python3-rpm-spec-0.14.1-1.cm2.noarch.rpm
[sudo] password for azaugg:
```
- Then imported the python module
```
azaugg@azaugg-ld77 [ ~/rpmbuild ]$ python3
Python 3.9.14 (main, Feb  8 2023, 03:39:52)
[GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyrpm.spec
>>>
azaugg@azaugg-ld77 [ ~/rpmbuild ]$
```
